src/social-network-provider/twitter.com/ui/fetch.ts

```diff
@@ -58,37 +67,40 @@ const registerUserCollector = () => {
 }
 
 const registerPostCollector = (self: SocialNetworkUI) => {
-    new MutationObserverWatcher(postsSelectors())
+    new MutationObserverWatcher(postsSelector())
         .useForeach((node, _, proxy) => {
-            const info = getEmptyPostInfoByElement({
-                get rootNode() {
-                    return proxy.current
-                },
-                rootNodeProxy: proxy,
-            })
-            const collectPostInfo = async () => {
-                const r = await postParser(node.querySelector<HTMLElement>('[data-testid="tweet"]')!)
-                info.postContent.value = r.content
-                const postBy = new PersonIdentifier(self.networkIdentifier, r.handle)
-                if (!info.postBy.value.equals(postBy)) {
-                    info.postBy.value = postBy
+            // noinspection JSUnnecessarySemicolon
+            ;(async () => {
+                const info = getEmptyPostInfoByElement({
+                    get rootNode() {
+                        return proxy.current
+                    },
+                    rootNodeProxy: proxy,
+                })
+                const collectPostInfo = async () => {
+                    const r = await postParser(node)
+                    if (!r.pid) return false
+                    info.postContent.value = r.content
+                    const postBy = new PersonIdentifier(self.networkIdentifier, r.handle)
+                    if (!info.postBy.value.equals(postBy)) {
+                        info.postBy.value = postBy
+                    }
+                    info.postID.value = r.pid
+                    uploadToService(r)
                 }
-                info.postID.value = r.pid
-                uploadToService(r)
-            }
-            collectPostInfo().then(() => {
+                await collectPostInfo()
                 info.postPayload.value = deconstructPayload(info.postContent.value, self.payloadDecoder)
                 info.postContent.addListener(newValue => {
                     info.postPayload.value = deconstructPayload(newValue, self.payloadDecoder)
                 })
                 // push to map. proxy used as a pointer here.
                 self.posts.set(proxy, info)
-            })
-            return {
-                onNodeMutation: collectPostInfo,
-                onTargetChanged: collectPostInfo,
-                onRemove: () => self.posts.delete(proxy),
-            }
+                return {
```

Comment:
By this way, this function is not able to provide callbacks like `onNodeMutation`

Reply:
aa606d345c6c6bc5927fb9b32d01277ace497d05